programs/drift_vaults/src/state/vault.rs

```diff
@@ -245,26 +440,28 @@ impl Vault {
         self.manager_net_deposits = self.manager_net_deposits.safe_add(amount.cast()?)?;
 
         self.total_shares = self.total_shares.safe_add(n_shares)?;
-        let vault_shares_after = self.total_shares.safe_sub(self.user_shares)?;
-
-        emit!(VaultDepositorRecord {
-            ts: now,
-            vault: self.pubkey,
-            depositor_authority: self.manager,
-            action: VaultDepositorAction::Deposit,
-            amount: 0,
-            spot_market_index: self.spot_market_index,
-            vault_equity_before: vault_equity,
-            vault_shares_before,
-            user_vault_shares_before,
-            total_vault_shares_before,
-            vault_shares_after,
-            total_vault_shares_after: self.total_shares,
-            user_vault_shares_after: self.user_shares,
-            profit_share: 0,
-            management_fee,
-            management_fee_shares,
-        });
+        let vault_shares_after = self.get_manager_shares(vault_protocol)?;
+
+        self.emit_vault_depositor_record(
+            VaultDepositorRecordParams {
+                ts: now,
+                action: VaultDepositorAction::Deposit,
+                amount: 0,
+                vault_equity_before: vault_equity,
+                vault_shares_before,
+                user_vault_shares_before,
+                total_vault_shares_before,
+                vault_shares_after,
+                manager_profit_share: 0,
+                management_fee: management_fee_payment,
+                management_fee_shares,
+            },
+            Some(VaultDepositorRecordProtocolParams {
```

Comment:
shouldn't this only be some if the vp is some? otherwise we'll always log the V1 record even for legacy vaults?